mapper/cumulocity/c8y_translator_lib/src/json.rs

```diff
@@ -36,9 +36,29 @@ pub fn from_thin_edge_json(input: &str) -> Result<String, CumulocityJsonError> {
 
 fn from_thin_edge_json_with_timestamp(
     input: &str,
-    default_timestamp: DateTime<FixedOffset>,
+    timestamp: DateTime<FixedOffset>,
 ) -> Result<String, CumulocityJsonError> {
-    let mut serializer = serializer::C8yJsonSerializer::new(default_timestamp);
+    let mut serializer = serializer::C8yJsonSerializer::new(timestamp);
+    let () = parse_str(input, &mut serializer)?;
+    Ok(serializer.into_string()?)
+}
+
+/// Converts from thin-edge Json to c8y_json for child device
+pub fn from_thin_edge_json_with_child(
+    input: &str,
+    child_id: &str,
+) -> Result<String, CumulocityJsonError> {
```

Comment:
I did refactoring for those 4 functions.

Public API (I didn't add `maybe_child_id` to Public APIs, especially I wanted to avoid a change in `from_thin_edge_json`, which is used in several places already)
- from_thin_edge_json(input: &str)
- from_thin_edge_json_with_child(input: &str,child_id: &str)

Private function
- from_thin_edge_json_with_timestamp(input: &str, timestamp: DateTime<FixedOffset>, maybe_child_id: Option<&str>)